src/fondant/pipeline.py

```diff
@@ -80,6 +82,7 @@ def __init__(
         self.arguments.setdefault("component_spec", self.component_spec.specification)
 
         self.number_of_gpus = number_of_gpus
+        self.node_pool_label = node_pool_label
         self.node_pool_name = node_pool_name
         self.p_volumes = p_volumes
         self.ephemeral_storage_size = ephemeral_storage_size
```

Comment:
Since both the label and the name depend on each other. Can you raise an error if one of them is not provided? 

```python
from fondant.exceptions import InvalidPipelineDefinition 
if bool(node_pool_label) != bool(node_pool_name):
        raise InvalidPipelineDefinition("Both 'node_pool_label' and 'node_pool_name' must be defined or both must be None.")
```

And can you also test that a valid error is returned by adding two example to `test_component_op`, one where the name is defined and the nodepool is not and the other way around: 

https://github.com/ml6team/fondant/blob/5b123bff57c933e3fbc24aac7a36f73bf4ddacd3/tests/test_pipeline.py#L35

Reply:
I added the error checking and test. Could you please check? @PhilippeMoussalli

Reply:
Thanks :)